TimberAPI/EntityLinkerSystem/EntityLinker.cs

```diff
@@ -0,0 +1,164 @@
+        protected static readonly ListKey<EntityLink> EntityLinksKey = new ListKey<EntityLink>(nameof(EntityLinks));
+
+        internal readonly List<EntityLink> _entityLinks = new List<EntityLink>();
+        public IReadOnlyCollection<EntityLink> EntityLinks { get; private set; }
+
+        public virtual void Awake()
+        {
+            EntityLinks = _entityLinks.AsReadOnly();
+        }
+
+        /// <summary>
+        /// Save existing Links
+        /// </summary>
+        /// <param name="entitySaver"></param>
+        public virtual void Save(IEntitySaver entitySaver)
+        {
+            IObjectSaver component = entitySaver.GetComponent(EntityLinkerKey);
+            component.Set(EntityLinksKey, EntityLinks);
+        }
+
+        /// <summary>
+        /// Load saves Links
+        /// </summary>
+        /// <param name="entityLoader"></param>
+        public virtual void Load(IEntityLoader entityLoader)
+        {
+            if (!entityLoader.HasComponent(EntityLinkerKey))
+            {
+                return;
+            }
+            IObjectLoader component = entityLoader.GetComponent(EntityLinkerKey);
+            if (component.Has(EntityLinksKey))
+            {
+                var links = component.Get(EntityLinksKey);
+                if (links == null)
+                {
+                    return;
+                }
+                var linkerLinks = links.Where(x => x.Linker == this).ToList();
+                _entityLinks.AddRange(linkerLinks);
+                foreach (var link in linkerLinks)
+                {
+                    PostCreateLink(link);
+                }
+            }
+        }
+
+        public virtual void OnEnterFinishedState()
+        {
+            //Nothing to do here?
+        }
+
+        /// <summary>
+        /// When entity is deleted, remove all links
+        /// </summary>
+        public virtual void OnExitFinishedState()
+        {
+            RemoveAllLinks();
+        }
+
+        /// <summary>
+        /// Creates a new link where this in the Linker
+        /// </summary>
+        /// <param name="linkee"></param>
+        public virtual void CreateLink(EntityLinker linkee)
+        {
+            if(linkee == this)
+            {
+                TimberAPIPlugin.Log.LogWarning($"Tried to link entity with itself. Stopped linking.");
+                return;
+            }
+            var link = new EntityLink(this, linkee);
+            AddLink(link);
+            PostCreateLink(link);
+        }
+
+        /// <summary>
+        /// Adds a new Link where this is the Linkee
+        /// </summary>
+        /// <param name="link"></param>
+        public virtual void AddLink(EntityLink link)
+        {
+            _entityLinks.Add(link);
+        }
+
+        /// <summary>
+        /// Add the Link on the Linkee
+        /// </summary>
+        /// <param name="link"></param>
+        public virtual void PostCreateLink(EntityLink link)
```

Comment:
Can we rename this?

Reply:
@hawkfalcon Yes. Gonna rename this and "PostDeleteLink" into "AddLinkInLinkee" and "DeleteLinkFromLinkee". Also changing them to private, because there shouldn't be a reason to call those methods from outside